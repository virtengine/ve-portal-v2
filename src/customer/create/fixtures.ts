export const user = {
  uuid: 'uuid',
  name: 'User',
  customer_permissions: [],
};

export const staff = {
  uuid: 'uuid',
  name: 'Staff',
  is_staff: true,
  customer_permissions: [],
};
