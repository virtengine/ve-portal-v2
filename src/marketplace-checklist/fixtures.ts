import { Checklist, ChecklistStats, Question, Answer, Category } from './types';

export const initCustomer = (props = {}): ChecklistStats => {
  return {
    uuid: '1',
    name: 'Alex',
    score: 0,
    latitude: 49.601984,
    longitude: 95.63666172955419,
    ...props,
  };
};

export const checklists: Checklist[] = [
  {
    uuid: '51525237e3bc40879c437f5d4d3be850',
    name: 'Best practices (2 questions)',
    description: '',
    questions_count: 2,
    category_name: 'Project checklist',
    category_uuid: '7fd33c0055d8453d84e4da5371dd8fab',
  },
  {
    uuid: '407296c2821345aab0f02a583293a476',
    name: 'Cyberhygiene (2 questions)',
    description: '',
    questions_count: 2,
    category_name: 'Project checklist',
    category_uuid: '7fd33c0055d8453d84e4da5371dd8fab',
  },
];

export const questions: Question[] = [
  {
    uuid: '554b3673d2f944299057143fa62be3ef',
    description: 'I assign PIDs to all my generated resources',
    solution: 'Please see RDA recommendations here and here',
    category_uuid: '6dfcdbe3c18241dc87526ef411bf6064',
    correct_answer: true,
  },
  {
    uuid: '4472d7df987644f39804cb7dbdbc2e6e',
    description: 'I preserve data in a storage with SLAs on preservation',
    solution: 'Find a service provider to use for keeping artifacts.',
    category_uuid: '6dfcdbe3c18241dc87526ef411bf6064',
    correct_answer: true,
  },
];

export const answers: Answer[] = [
  {
    question_uuid: '554b3673d2f944299057143fa62be3ef',
    value: false,
  },
  {
    question_uuid: '4472d7df987644f39804cb7dbdbc2e6e',
    value: true,
  },
];

export const category: Category = {
  uuid: '7fd33c0055d8453d84e4da5371dd8fab',
  icon: null,
  name: 'Project checklist',
  checklists_count: 2,
};
