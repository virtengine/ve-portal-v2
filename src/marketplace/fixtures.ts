export const customer = {
  uuid: '6983ac22f2bb469189311ab21e493359',
  is_service_provider: true,
};
