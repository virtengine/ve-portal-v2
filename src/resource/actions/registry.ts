class ActionConfiguration {
  private resources = {};

  register(type, config: any) {
    this.resources[type] = config;
  }

  getActions(type) {
    return this.resources[type];
  }
}

export const ActionRegistry = new ActionConfiguration();
