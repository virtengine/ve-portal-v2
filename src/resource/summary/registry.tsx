import { ComponentType } from 'react';

const registry = {};

export const register = (
  type: string,
  component: ComponentType<any>,
  useDefaultWrapper = true,
) => {
  registry[type] = {
    component,
    useDefaultWrapper,
  };
};

export const get = (type: string) => {
  return registry[type];
};
