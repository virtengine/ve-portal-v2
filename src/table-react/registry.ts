import { TableOptions } from './types';

const registry = {};

export const registerTable = (options: TableOptions) => {
  registry[options.table] = options;
};

export const getTableOptions: (name: string) => TableOptions = name => {
  if (!registry[name]) {
    throw Error(`Loader for datatable ${name} is not found.`);
  }

  return registry[name];
};
