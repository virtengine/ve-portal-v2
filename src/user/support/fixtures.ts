export const staffUser = {
  is_support: false,
  is_staff: true,
};

export const supportUser = {
  is_support: true,
  is_staff: false,
};

export const supportAndStaffUser = {
  is_support: true,
  is_staff: true,
};

export const ordinaryUser = {
  is_support: false,
  is_staff: false,
  civil_number: '14124h12kj4h',
};

export const userWithFacebookId = {
  registration_method: 'facebook',
};
