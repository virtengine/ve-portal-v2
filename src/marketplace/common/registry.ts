import * as React from 'react';
import { Omit } from 'react-redux';

import { Offering, Attribute, OrderItemDetailsProps, OfferingConfigurationFormProps } from '@waldur/marketplace/types';

const REGISTRY: {[key: string]: Omit<OfferingConfiguration, 'type'>} = {};

interface OfferingConfiguration<AttributesType = any, RequestPaylodType = any> {
  type: string;
  component: React.ComponentType<OfferingConfigurationFormProps>;
  detailsComponent?: React.ComponentType<OrderItemDetailsProps>;
  checkoutSummaryComponent?: any;
  serializer?: (attributes: AttributesType, offering: Offering) => RequestPaylodType;
  limitSerializer?: (limit: Record<string, number>) => Record<string, number>;
  limitParser?: (limit: Record<string, number>) => Record<string, number>;
  label: string;
  showOptions?: boolean;
  showComponents?: boolean;
  providerType?: string;
  attributes?(): Attribute[];
  disableOfferingCreation?: boolean;
  schedulable?: boolean;
}

export interface Option {
  value: string;
  label: string;
}

export function registerOfferingType(config: OfferingConfiguration) {
  const {type, ...rest} = config;
  REGISTRY[type] = rest;
}

export function getFormComponent(offeringType) {
  return REGISTRY.hasOwnProperty(offeringType) && REGISTRY[offeringType].component;
}

export function getDetailsComponent(offeringType) {
  return REGISTRY.hasOwnProperty(offeringType) && REGISTRY[offeringType].detailsComponent;
}

export function getFormSerializer(offeringType) {
  return REGISTRY.hasOwnProperty(offeringType) && REGISTRY[offeringType].serializer || (x => x);
}

export function getFormLimitSerializer(offeringType) {
  return REGISTRY.hasOwnProperty(offeringType) && REGISTRY[offeringType].limitSerializer || (x => x);
}

export function getFormLimitParser(offeringType) {
  return REGISTRY.hasOwnProperty(offeringType) && REGISTRY[offeringType].limitParser || (x => x);
}

export function getCheckoutSummaryComponent(offeringType) {
  return REGISTRY.hasOwnProperty(offeringType) && REGISTRY[offeringType].checkoutSummaryComponent;
}

export function getOfferingTypes(): Option[] {
  const keys = Object.keys(REGISTRY).filter(key => !REGISTRY[key].disableOfferingCreation);
  return keys.map(key => ({
    value: key,
    label: REGISTRY[key].label,
  }));
}

export function showOfferingOptions(offeringType) {
  return REGISTRY.hasOwnProperty(offeringType) && REGISTRY[offeringType].showOptions;
}

export function isOfferingTypeSchedulable(offeringType) {
  return REGISTRY.hasOwnProperty(offeringType) && REGISTRY[offeringType].schedulable;
}

export function showComponentsList(offeringType) {
  return REGISTRY.hasOwnProperty(offeringType) && REGISTRY[offeringType].showComponents;
}

export function getProviderType(offeringType) {
  return REGISTRY.hasOwnProperty(offeringType) && REGISTRY[offeringType].providerType;
}

export function getLabel(offeringType) {
  return REGISTRY.hasOwnProperty(offeringType) && REGISTRY[offeringType].label;
}

export function getAttributes(offeringType) {
  const func = REGISTRY.hasOwnProperty(offeringType) && REGISTRY[offeringType].attributes;
  return func ? func() : [];
}
