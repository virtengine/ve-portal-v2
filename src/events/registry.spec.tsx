import { event } from './fixtures';
import { EventRegistry } from './registry';

describe('Event registry', () => {
  let registry: EventRegistry;

  beforeEach(() => {
    registry = new EventRegistry();
  });

  it('returns event message as is by default', () => {
    expect(registry.formatEvent(event)).toBe(event.message);
  });

  it('uses custom formatter if possible', () => {
    registry.registerGroup({
      title: 'User events',
      events: [
        {
          key: event.event_type,
          title: 'User {user} authenticated successfully.',
          formatter: (e) =>
            `User ${e.user_full_name} authenticated successfully.`,
        },
      ],
    });
    expect(registry.formatEvent(event)).toMatchSnapshot();
  });

  it('uses custom context if possible', () => {
    registry.registerGroup({
      title: 'User events',
      context: (e) => ({
        user_link: <a href={e.user_uuid}>{e.user_full_name}</a>,
      }),
      events: [
        {
          key: event.event_type,
          title: 'User {user_link} authenticated successfully.',
        },
      ],
    });
    expect(registry.formatEvent(event)).toMatchSnapshot();
  });

  it('uses event context as is by default', () => {
    registry.registerGroup({
      title: 'User events',
      events: [
        {
          key: event.event_type,
          title: 'User {user_full_name} authenticated successfully.',
        },
      ],
    });
    expect(registry.formatEvent(event)).toMatchSnapshot();
  });
});
