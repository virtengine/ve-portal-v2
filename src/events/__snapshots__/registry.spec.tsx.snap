// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event registry uses custom context if possible 1`] = `
<React.Fragment>
  <React.Fragment>
    User 
  </React.Fragment>
  <React.Fragment>
    <a
      href="a45c8927e176455781f63e7c7baf9567"
    >
      Alice Lebowski
    </a>
  </React.Fragment>
  <React.Fragment>
     authenticated successfully.
  </React.Fragment>
</React.Fragment>
`;

exports[`Event registry uses custom formatter if possible 1`] = `"User Alice Lebowski authenticated successfully."`;

exports[`Event registry uses event context as is by default 1`] = `
<React.Fragment>
  <React.Fragment>
    User 
  </React.Fragment>
  <React.Fragment>
    Alice Lebowski
  </React.Fragment>
  <React.Fragment>
     authenticated successfully.
  </React.Fragment>
</React.Fragment>
`;
